### DISCUSSION TITLE
Regex usage in search section

### DISCUSSION THREAD

**Discussion by hidayetkurtulmus** [netbox-community/netbox#15149] [Q&A]
Is it possible to use regex in search section?

**Comment by wirnico** [comment]
Hey there,
did you ever get an answer? I see Regex is possible in the main search but not in the quick search under virtual machines for example. There it would be very helpful for me to select the result and edit multiple machines or devices.
Thanks, Nico

**Comment by DonZeptero** [comment]
Hello,

i ran into the same issue and i solve it this way:

Depending on the version of netbox you are using you need to enable the FILTER_CHAR_BASED_LOOKUP_MAP to enable search for regex, in the latest version it is already present.
Here is the pull request:
[update lookup map](https://github.com/netbox-community/netbox/pull/19912/files)

If the lookup expressions are availaible, you can change the search behavior for every type by replacing the query filter for the field in the filtersets.py.

Here is an example for devices (netbox/dcim/filtersets.py), the patch is valid for latest version. So you need to update the search function of class DeviceFilterSet.
```
diff --git a/netbox/dcim/filtersets.py b/netbox/dcim/filtersets.py
index 90a9993c2..e591823aa 100644
--- a/netbox/dcim/filtersets.py
+++ b/netbox/dcim/filtersets.py
@@ -1192,7 +1192,7 @@ class DeviceFilterSet(
         if not value.strip():
             return queryset
         return queryset.filter(
-            Q(name__icontains=value) |
+            Q(name__regex=value) |
             Q(serial__icontains=value.strip()) |
             Q(inventoryitems__serial__icontains=value.strip()) |
             Q(asset_tag__icontains=value.strip()) |

```
